pkg/compactor/compactor.go

```diff
@@ -198,6 +216,14 @@ func (cfg *Config) Validate() error {
 		return errInvalidShardingStrategy
 	}
 
+	if cfg.ShardingStrategy == util.ShardingStrategyShuffle {
+		if !cfg.ShardingEnabled {
+			return errShardingRequired
```

Comment:
I am not sure I agree we should throw this error only for shuffle sharding but not the default sharding. I also understand that the sharding strategy is set to "default" by default. So I think what should be done is this: 

Since the code block seems to want to just validate the shard size for shuffle sharding, what about doing: 

```
if (shardingIsEnabled && isShuffleSharding) {
   validateShardSize
} 
```
My thinking is that if `ShardingEnable=true` and `ShardingStrategy=defualt` doesn't fail validation, then `ShardingEnable=true` and `ShardingStrategy=shuffle-sharding` should not fail validation too.